app/src/molecules/Modal/OnDeviceDisplay/Modal.tsx

```diff
@@ -0,0 +1,73 @@
+import * as React from 'react'
+import {
+  BORDERS,
+  COLORS,
+  Flex,
+  SPACING,
+  DIRECTION_COLUMN,
+  JUSTIFY_CENTER,
+} from '@opentrons/components'
+import { BackgroundOverlay } from '../../BackgroundOverlay'
+import { ModalHeader } from './ModalHeader'
+
+import type { ModalHeaderProps, ModalSize } from './types'
+
+interface ModalProps {
+  //  clicking anywhere outside of the modal closes it
+  onOutsideClick: React.MouseEventHandler
+  children: React.ReactNode
+  //  for small, medium, or large modal sizes, medium by default
+  modalSize?: ModalSize
+  header?: ModalHeaderProps
+}
+export function Modal(props: ModalProps): JSX.Element {
+  const { modalSize = 'medium', onOutsideClick, children, header } = props
+
+  let modalWidth: string = '45.625rem'
+  switch (modalSize) {
+    case 'small': {
+      modalWidth = '32.375rem'
+      break
+    }
+    case 'large': {
+      modalWidth = '60rem'
+      break
+    }
+  }
+  return (
+    <BackgroundOverlay
+      onClick={e => {
+        e.stopPropagation()
+        onOutsideClick(e)
+      }}
+      justifyContent={JUSTIFY_CENTER}
+    >
+      <Flex
+        backgroundColor={COLORS.white}
+        width={modalWidth}
+        max-height="32.5rem"
```

Comment:
Just saw Storybook on sandbox and the large modal's height is `630px`
```suggestion
        maxHeight="32.5rem"
```

Reply:
oops good catch!